velox/exec/tests/SortBufferTest.cpp

```diff
@@ -0,0 +1,163 @@
+      {{"c0", BIGINT()},
+       {"c1", INTEGER()},
+       {"c2", SMALLINT()},
+       {"c3", REAL()},
+       {"c4", DOUBLE()},
+       {"c5", VARCHAR()}});
+  std::vector<column_index_t> sortColumnIndices_{4, 1};
+  std::vector<CompareFlags> sortCompareFlags_{
+      {true, true, false, false},
+      {true, true, false, false}};
+
+  std::shared_ptr<memory::MemoryPool> rootPool_{
+      memory::defaultMemoryManager().addRootPool("SortBufferTest")};
+  std::shared_ptr<memory::MemoryPool> pool_{
+      rootPool_->addLeafChild("SortBufferTest")};
+  tsan_atomic<bool> nonReclaimableSection_{false};
+};
+
+TEST_F(SortBufferTest, basic) {
+  auto sortBuffer = new SortBuffer(
+      inputType_,
+      sortColumnIndices_,
+      sortCompareFlags_,
+      pool_.get(),
+      &nonReclaimableSection_);
+
+  RowVectorPtr data = makeRowVector(
+      {makeFlatVector<int64_t>({1, 2, 3, 4, 5}),
+       makeFlatVector<int32_t>({5, 4, 3, 2, 1}), // sorted-2 column
+       makeFlatVector<int16_t>({1, 2, 3, 4, 5}),
+       makeFlatVector<float>({1.1, 2.2, 3.3, 4.4, 5.5}),
+       makeFlatVector<double>({1.1, 2.2, 2.2, 5.5, 5.5}), // sorted-1 column
+       makeFlatVector<std::string>(
+           {"hello", "world", "today", "is", "great"})});
+
+  sortBuffer->addInput(data);
+  sortBuffer->noMoreInput();
+  auto output = sortBuffer->getOutput();
+  ASSERT_EQ(output->size(), 5);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(0), 5);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(1), 3);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(2), 4);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(3), 1);
+  ASSERT_EQ(output->childAt(1)->asFlatVector<int32_t>()->valueAt(4), 2);
+}
+
+TEST_F(SortBufferTest, batchOutput) {
+  auto sortBuffer = new SortBuffer(
+      inputType_,
+      sortColumnIndices_,
+      sortCompareFlags_,
+      pool_.get(),
+      &nonReclaimableSection_);
+
+  vector_size_t numRows = 35000;
+  RowVectorPtr data = makeRowVector(
+      {makeFlatVector<int64_t>(numRows, [](auto row) { return row; }),
+       makeFlatVector<int32_t>(
+           numRows, [](auto row) { return row; }), // sorted-2 column
+       makeFlatVector<int16_t>(numRows, [](auto row) { return row; }),
+       makeFlatVector<float>(numRows, [](auto row) { return row; }),
+       makeFlatVector<double>(
+           numRows, [](auto row) { return row; }), // sorted-1 column
+       makeFlatVector<std::string>(numRows)});
+
+  sortBuffer->addInput(data);
+  sortBuffer->noMoreInput();
+
+  // Verify output batch: batchSizeLimit_ is 10000, thus expect 4 batches.
+  int expectedBatchCount = 4;
+  for (int i = 0; i < expectedBatchCount; ++i) {
+    auto output = sortBuffer->getOutput();
+    // Verify sorting order
+    ASSERT_EQ(
+        output->childAt(1)->asFlatVector<int32_t>()->valueAt(0), i * 10000);
+    ASSERT_NE(output, nullptr);
+  }
+  ASSERT_EQ(sortBuffer->getOutput(), nullptr);
+}
```

Comment:
I think it is better to have a test utility here
```
RowVectorPtr buildExpectedSortResult(std::vector<> inputVectors, sort flags)
```
Then use this to verify the results that we yield from the sort buffer result?

Reply:
but what is the implementation of this test utility? there is no utilities we can use in velox that can do sorting for inputVectors by sort keys?